pkg/controller/master/setup.go

```diff
@@ -65,15 +66,19 @@ func register(ctx context.Context, management *config.Management, options config
 func Setup(ctx context.Context, server *server.Server, controllers *server.Controllers, options config.Options) error {
 	scaled := config.ScaledWithContext(ctx)
 
+	startupComplete := make(chan bool, 1)
 	go leader.RunOrDie(ctx, "", "harvester-controllers", controllers.K8s, func(ctx context.Context) {
 		if err := register(ctx, scaled.Management, options); err != nil {
 			panic(err)
 		}
 		if err := scaled.Management.Start(options.Threadiness); err != nil {
 			panic(err)
 		}
-		<-ctx.Done()
+		startupComplete <- true
 	})
 
+	logrus.Debug("waiting for controller boot to complete")
+	<-startupComplete
```

Comment:
question:

will all cached data be ready here ?

the leader election is backed by a special CRD object, I suspect this object is ready, does not mean all other CRD objects are ready, please correct me, thanks.

Reply:
yeah they are ready. only problem as we have found is that it blocks steve from starting up on all nodes in a HA cluster. So i need to fine tune it more or switch helm chart lookup to using the client rather than the cache.

Reply:
i have changed the controller to use the client and avoid this issue for now.